loader/loader_test.go

```diff
@@ -238,6 +245,85 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		})
 	})
 
+	When("calling GetPreviousRelease", func() {
+		var release1, release2, release3 *v1alpha1.Release
+
+		BeforeEach(func() {
+			now := time.Now()
+
+			release1 = &v1alpha1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "release1",
+					Namespace:         "default",
+					CreationTimestamp: metav1.Time{Time: now.Add(-3 * time.Hour)},
+				},
+				Spec: v1alpha1.ReleaseSpec{
+					ReleasePlan: "release-plan-1",
+					Snapshot:    "snapshot-1",
+				},
+			}
+			release2 = &v1alpha1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "release2",
+					Namespace:         "default",
+					CreationTimestamp: metav1.Time{Time: now.Add(-2 * time.Hour)},
+				},
+				Spec: v1alpha1.ReleaseSpec{
+					ReleasePlan: "release-plan-2",
+					Snapshot:    "snapshot-2",
+				},
+			}
+			release3 = &v1alpha1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "release3",
+					Namespace:         "default",
+					CreationTimestamp: metav1.Time{Time: now.Add(-1 * time.Hour)},
+				},
+				Spec: v1alpha1.ReleaseSpec{
+					ReleasePlan: "release-plan-3",
+					Snapshot:    "snapshot-3",
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, release1)).To(Succeed())
+			Expect(k8sClient.Create(ctx, release2)).To(Succeed())
+			Expect(k8sClient.Create(ctx, release3)).To(Succeed())
+
+			// Ensure resources are created
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: release1.Name, Namespace: release1.Namespace}, release1)
+			}, timeout, interval).Should(Succeed())
+
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: release2.Name, Namespace: release2.Namespace}, release2)
+			}, timeout, interval).Should(Succeed())
+
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: release3.Name, Namespace: release3.Namespace}, release3)
+			}, timeout, interval).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, release1)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, release2)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, release3)).To(Succeed())
+		})
+
+		It("returns the previous release if it exists", func() {
+			returnedObject, err := loader.GetPreviousRelease(ctx, k8sClient, release3)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(BeNil())
+			Expect(returnedObject.Name).To(Equal(release2.Name))
```

Comment:
It is failing here because your logic is working (I suppose). The `release` release is newer than `release{1-3}` as all of them are 1 to 3 hours older. Try  to set the CreationTimestamp using difference of seconds.

Reply:
or just create them syncrhonously.

Reply:
Thanks. I will continue as discussed offline.